docs/next/guides/sidebar.js

```diff
@@ -1,189 +1,189 @@
 const gettingStartedItems = [
-  'guides/getting-started/introduction',
-  'guides/getting-started/demo',
-  'guides/getting-started/installation',
-  'guides/getting-started/binding-to-data',
-  'guides/getting-started/saving-data',
-  'guides/getting-started/setting-options',
-  'guides/getting-started/grid-size',
-  'guides/getting-started/events-and-hooks',
-  'guides/getting-started/license-key',
+  { link: 'guides/getting-started/introduction' },
```

Comment:
Maybe I'm naive, but it seems to me that a simpler solution would be to filter the items directly in the `sidebar.js`. Then you would need no change to inform VuePress about the skipped pages:

```js
const filterOnlyFor = framework => element => element.onlyFor === undefined || element.onlyFor === framework;

let gettingStartedItems = [
  { link: 'guides/getting-started/introduction' },
  { link: 'guides/getting-started/demo' },
  { link: 'guides/getting-started/installation' },
  { link: 'guides/getting-started/binding-to-data' },
  { link: 'guides/getting-started/saving-data' },
  { link: 'guides/getting-started/setting-options' },
  { link: 'guides/getting-started/grid-size' },
  { link: 'guides/getting-started/events-and-hooks' },
  { link: 'guides/getting-started/license-key', onlyFor: 'react' },
].filter(filterOnlyFor('javascript')); // returns everything except onlyFor: 'react'
```

Reply:
At first, it looked for me promising. I thought about exporting some method using factory design pattern. However, after further investigation I would say that there are some disadvantages of doing it.

It seems that we won't be able for excluding sites from the search without getting knowledge which one of them should be filtered out. Thus, we should have 2 methods (or one, even more sophisticated, which may be anti-pattern):

- for filtering elements basing on the handled framework (framework could be passed as function argument)
- for getting **filtered out elements** (traversation of nested structure should be done)

Thus, there should be an extra, quite sopisticated code which would be duplicated for next `sidebar.js` files or an extra util (hard to say where to put it - in the main directory?). It seems that files have been designed as simple configuration object, so I don't like this idea.